src/graphene.cpp

```diff
@@ -792,6 +792,91 @@ double CGrapheneBlockData::average(std::map<int64_t, uint64_t> &map)
     return (double)accum / map.size();
 }
 
+double CGrapheneBlockData::computTotalBandwidthSavingsInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_graphenestats)
+{
+    AssertLockHeld(cs_graphenestats);
+
+    return double(nOriginalSize() - nGrapheneSize() - nTotalMemPoolInfoBytes());
+}
+
+double CGrapheneBlockData::compute24hInboundCompressionInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_graphenestats)
+{
+    AssertLockHeld(cs_graphenestats);
+
+    expireStats(mapGrapheneBlocksInBound);
+
+    double nCompressionRate = 0;
+    uint64_t nGrapheneSizeTotal = 0;
+    uint64_t nOriginalSizeTotal = 0;
+    for (std::map<int64_t, std::pair<uint64_t, uint64_t> >::iterator mi = mapGrapheneBlocksInBound.begin();
+         mi != mapGrapheneBlocksInBound.end(); ++mi)
+    {
+        nGrapheneSizeTotal += (*mi).second.first;
+        nOriginalSizeTotal += (*mi).second.second;
+    }
+    // We count up the outbound CMemPoolInfo sizes. Outbound CMemPoolInfo sizes go with Inbound graphene blocks.
+    uint64_t nOutBoundMemPoolInfoSize = 0;
+    for (std::map<int64_t, uint64_t>::iterator mi = mapMemPoolInfoOutBound.begin(); mi != mapMemPoolInfoOutBound.end();
+         ++mi)
+    {
+        nOutBoundMemPoolInfoSize += (*mi).second;
+    }
+
+    if (nOriginalSizeTotal > 0)
+        nCompressionRate = 100 - (100 * (double)(nGrapheneSizeTotal + nOutBoundMemPoolInfoSize) / nOriginalSizeTotal);
+
+    return nCompressionRate;
+}
+
+double CGrapheneBlockData::compute24hOutboundCompressionInternal() EXCLUSIVE_LOCKS_REQUIRED(cs_graphenestats)
```

Comment:
Can you further DRY that with the inbound function above? Seems like they share a lot of code still...

Reply:
Will do.

Reply:
Fixed